### ISSUE TITLE
pretrinketing bug

### ISSUE THREAD

**Issue by RandomXD** [dalaranwow/dalaran-wow#5637]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
If you equip trinks with an "use" effect (example eot trinket which you can proc yourself) you can use it before a fight starts then equip your normal trinket and the bonus from it will still be up even tho you unequipped the trinket
**Expected behaviour**: Tell us what should happen instead
pretrinketing shouldn't be a thing, once you remove the trinket the buff from it should be removed too
**Steps to reproduce the problem**:

1. get a trinket with "use" like the eot one or the eof caster one
2. equip it and use it
3. remove it and equip your normal trinket the buff will still be up

**Include proofs for this behaviour**

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by mercred** [comment]
Tested it yesterday and the buff from the trinket got removed as soon as another trinket was equiped. Maybe this is a bug with a particular trinket? Please provide more details.

**Comment by LoveDalaranWow** [comment]
The bug still exists and it's abused by most people raiding regularly.

**Comment by Chenocide** [comment]
This is still a problem with relics (sigils, librams, totems and idols). Buffs from relics should be removed if the item is taken off.

**Comment by mercred** [comment]
@Lothloryen according to my guildies pre-trinketing Nevermelting Ice Crystal still works and doesn't even require a macro for it

**Comment by Bajeru** [comment]
@mercred that trinket is supposed to be like that.

**Comment by mercred** [comment]
@Bajeru please help finding source for it being intended. I checked EJ Shadow priest guide(as I think it should be part of the opener for them if its intended) and wowhead as well as some random googling but couldn't find anything.
